synapse/rest/client/room_batch.py

```diff
@@ -131,6 +131,14 @@ async def on_POST(
             prev_event_ids_from_query
         )
 
+        if not auth_event_ids:
+            raise SynapseError(
+                HTTPStatus.BAD_REQUEST,
+                "No auth events found for given prev_event query parameter. The prev_event=%s probably does not exist."
+                % prev_event_ids_from_query,
+                errcode=Codes.INVALID_PARAM,
+            )
+
         state_event_ids_at_start = []
         # Create and persist all of the state events that float off on their own
         # before the batch. These will most likely be all of the invite/member
```

Comment:
On line 213 below here, we can probably disconnect the floating state chain from the `insertion` event. Related discussion https://github.com/matrix-org/synapse/pull/11243/files#discussion_r760841874

Remove these lines:

```py
if len(state_event_ids_at_start):
     prev_event_ids = [state_event_ids_at_start[-1]]
```

It should be fine for the `state_groups` which we can verify with this regression test https://github.com/matrix-org/synapse/pull/11487 and Complement tests to ensure everything works normally.